web/app/scripts/utils/date-utils.js

```diff
@@ -0,0 +1,87 @@
+(function() {
+    'use strict';
+
+    /* ngInject */
+    function DateUtils(LanguageState) {
+        var languageMap = {
+            'ar-sa': {
+                language: 'ar',
+                calendar: 'ummalqura',
+                formats: {
+                    'short': 'M Y',
+                    'long': 'd MM, Y',
+                    'numeric': ' - dd/mm/yyyy'
+                }
+            },
+            'en-us': {
+                language: 'en',
+                calendar: 'gregorian',
+                formats: {
+                    'short': 'M Y',
+                    'long': 'MM d, Y',
+                    'numeric': 'mm/dd/yyyy - '
+                }
+            },
+            'exclaim': {
+                language: 'en',
+                calendar: 'gregorian',
+                formats: {
+                    'short': '\'!\'M Y',
+                    'long': '\'!\'MM d, Y',
+                    'numeric': '\'!\'mm/dd/yyyy - '
+                }
+            }
+        };
+
+        var module = {
+            getLocalizedDateString: getLocalizedDateString
+        };
+        return module;
+
+        function getLocalizedDateString(date, format, includeTime) {
+            var language = LanguageState.getSelected().id;
+            if (!language) {
+                language = 'exclaim';
+            }
+            var isRtl = LanguageState.getSelected().rtl;
+            var conversion = languageMap[language];
+            var parsedDate = $.calendars.instance()
+                .parseDate('mm/dd/yyyyy', date.toLocaleDateString()).toJSDate();
+            var convertedDate = $.calendars.instance(conversion.calendar, conversion.language)
+                .fromJSDate(parsedDate);
+            var datestring = convertedDate.formatDate(conversion.formats[format], convertedDate);
+            if(includeTime) {
+                var minutes = date.getMinutes() + '';
```

Comment:
Concluded that while it's more concise, it isn't accurate (12hr clock vs 24hr, roman vs arabic numerals)